serviceregistration/consul/consul_service_registration.go

```diff
@@ -229,6 +231,20 @@ func (c *serviceRegistration) merge(conf map[string]string) error {
 		c.logger.Debug("config service_tags set", "service_tags", tags)
 	}
 
+	// Get user-defined meta tags to attach to the registered service name
+	metaTagsJSON, ok := conf["service_meta"]
+	var metaTags = map[string]string{}
+	if ok {
+		if err := json.Unmarshal([]byte(metaTagsJSON), &metaTags); err != nil {
+			return errors.New("service tags must be a dictionary of string keys and values")
+		}
+		metaTags["external-source"] = metaExternalSource
```

Comment:
We should do this irrespective of `if ok`, since otherwise there will be a behaviour change as part of this PR. Previously, the tag would have been set, but now, it won't be set unless I provide my own meta tags.

Reply:
To put it another way: if I don't provide my own `service_meta`, it should still be:
```
map[string]string{
	"external-source": metaExternalSource,
},
```

Reply:
I'd suggest moving this line until after the `if ok`, that way the behaviour is preserved.

Reply:
good catch, fixed!